DocumentDataAPI/DocumentDataAPI/Data/Repositories/NpgDataSourceRepository.cs

```diff
@@ -0,0 +1,83 @@
+using System.Data;
+using Dapper;
+using Dapper.FluentMap.Mapping;
+using DocumentDataAPI.Data.Mappers;
+using DocumentDataAPI.Models;
+
+namespace DocumentDataAPI.Data.Repositories;
+
+public class NpgDataSourceRepository : IDataSourceRepository
+{
+    private readonly IDbConnectionFactory _connectionFactory;
+    private readonly ILogger<NpgDataSourceRepository> _logger;
+
+    // Column name helpers
+    private readonly EntityMap<DataSourceModel> _map = new DataSourceMap();
+    private string IdColumn => _map.GetMappedColumnName(nameof(DataSourceModel.Id));
+    private string NameColumn => _map.GetMappedColumnName(nameof(DataSourceModel.Name));
+    private string DataSourceIdColumn => new DocumentMap().GetMappedColumnName(nameof(DocumentModel.DataSourceId));
+
+    public NpgDataSourceRepository(IDbConnectionFactory connectionFactory, ILogger<NpgDataSourceRepository> logger)
+    {
+        _connectionFactory = connectionFactory;
+        _logger = logger;
+    }
+
+    public async Task<DataSourceModel?> Get(long id)
+    {
+        _logger.LogDebug("Retrieving Source with id {id} from database", id);
+        using IDbConnection con = _connectionFactory.CreateConnection();
+        return await con.QueryFirstOrDefaultAsync<DataSourceModel>($"select * from data_sources where {IdColumn}=@Id",
+            new { id });
+    }
+
+    public async Task<IEnumerable<DataSourceModel>> GetAll()
+    {
+        _logger.LogDebug("Retrieving all Sources from database");
+        using IDbConnection con = _connectionFactory.CreateConnection();
+        return await con.QueryAsync<DataSourceModel>("select * from data_sources");
+    }
+
+    public async Task<int> Add(DataSourceModel entity)
+    {
+        _logger.LogDebug("Adding Source with id {Id} to database", entity.Id);
+        _logger.LogTrace("Source: {Source}", entity);
+        using IDbConnection con = _connectionFactory.CreateConnection();
+        return await con.ExecuteAsync($"insert into data_sources({NameColumn}) values (@Name)",
```

Comment:
@Sarmisuper 
Har forsøgt at tilføje kolonnernes navne som private get-only properties her. Må indrømme at jeg ikke er helt tilfreds med at vi bruger string interpolation her, det går stik imod al fornuft ift. sikkerhed. På den anden side, så er det properties der kun findes internt, og ikke kommer fra brugeren, så man kan godt argumentere for det.
Anyways, vi må lige blive enige i gruppen om hvorvidt vi skal bruge det her eller ej.

Reply:
Ja som du siger så bruger vi kun string interpolation til internt, så det burde vel ikke være et problem. Hvis der er nogen som har mulighed for at udnytte vores brug af string interpolation her så må de vel have access til at gøre så meget værre ting